compilerlibs/Makefile.compilerlibs

```diff
@@ -392,11 +392,15 @@ TOPLEVEL_SHARED_ARTEFACTS = $(TOPLEVEL_SHARED_MLIS) $(TOPLEVEL_SHARED_CMIS)
 
 $(addprefix toplevel/byte/, $(TOPLEVEL_SHARED_CMIS)):\
 toplevel/byte/%.cmi: toplevel/%.cmi
-	cp $< toplevel/$*.mli $(@D)
+	cp $< $(SRCDIR)/toplevel/$*.mli $(@D)
+
+toplevel/byte/topeval.cmo toplevel/native/topeval.cmx: OC_COMMON_COMPFLAGS += -cmi-file toplevel/byte/topeval.cmi
+toplevel/byte/trace.cmo toplevel/native/trace.cmx: OC_COMMON_COMPFLAGS += -cmi-file toplevel/byte/trace.cmi
+toplevel/byte/topmain.cmo toplevel/native/topmain.cmx: OC_COMMON_COMPFLAGS += -cmi-file toplevel/byte/topmain.cmi
```

Comment:
Why do we need this stuff?

Reply:
Explained [below](https://github.com/ocaml/ocaml/pull/12203#issuecomment-1522832365).